src/components/operations/BuyCoinsOperation.tsx

```diff
@@ -0,0 +1,177 @@
+import ActiveOperationButton from 'components/form/ActiveOperationButton';
+import AssetImage from 'components/ui/AssetImage';
+import Separator from 'components/ui/Separator';
+import React, {useRef, useState} from 'react';
+import {
+  FlatList,
+  Linking,
+  StyleSheet,
+  Switch,
+  Text,
+  useWindowDimensions,
+  View,
+} from 'react-native';
+import {connect, ConnectedProps} from 'react-redux';
+import ShoppingCartIconBlack from 'src/assets/wallet/icon_shopping_cart_black.svg';
+import {BuyCoinType} from 'src/enums/operations.enum';
+import {BuyCoinsListItem} from 'src/reference-data/buy-coins-list-item.list';
+import {RootState} from 'store';
+import {translate} from 'utils/localize';
+import Operation from './Operation';
+
+export type BuyCoinsprops = {
+  currency: BuyCoinType;
+};
+type Props = PropsFromRedux & BuyCoinsprops;
+
+const BuyCoinsOperation = ({user, currency}: Props) => {
+  const flatListRef = useRef();
+  const [isEnabled, setIsEnabled] = useState(
+    currency === BuyCoinType.BUY_HDB ? true : false,
```

Comment:
HIVE and HBD choices for buying are inverted. Transak should show for HIVE not HBD

Reply:
Done.